sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TruncatableTable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import org.apache.spark.annotation.Evolving;
+
+/**
+ * Represents a table which can be atomically truncated.
+ */
+@Evolving
+public interface TruncatableTable extends Table {
+  /**
+   * Truncate a table by removing all rows from the table atomically.
+   * If the table supports partitions, the method removes all existing partitions.
+   *
+   * @return true if a table was truncated successfully otherwise false
+   *
+   * @since 3.2.0
+   */
+  boolean truncateTable();
```

Comment:
Why include "table" in the method name? `table.truncate()` seems clear enough to me.

Reply:
Actually, this seems to mirror the other uses so I'm okay with it. I'd probably remove it but it seems okay to leave as is if you feel strongly about it.

Reply:
I just followed naming convention (maybe informal one) in other interfaces. For instance, if a table implements `TruncatableTable`, `SupportsPartitionManagement` and `SupportsAtomicPartitionManagement`, we have 3 methods in the same namespace:
- truncatePartition()
- truncatePartitions()
- and truncate()

In that case, maybe it is better to name this method as `truncateTable()` which can highlight that it is applied to entire table.